### PR TITLE
refactor: Extract bash functions into imported modules

### PR DESCRIPTION
**Pull request by mflendrich**
This PR creates 2 new files: `/lib/functions.sh` and `/lib/binaries.sh` to be `source`d from the main script.
This serves two purposes:

1. (main motivation) makes it possible to reuse these functions in other quickstart repos (by `git subtree` or a different yet undefined mechanism)
2. (nice side effect) separates the quickstart logic from primitive operation implementations for added visibility and maintainability

The implementations are unchanged.

Tested manually - the quickstart continues to work.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -1,328 +1,15 @@
 #!/usr/bin/env bash
 
+. $(dirname $0)/lib/functions.sh
+. $(dirname $0)/lib/binaries.sh
```
Comment (stealthybox) [inline setup.sh]: We can use some bash features to make this safer:
```bash
unset CD_PATH
SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
cd "${SCRIPT_DIR}" || exit 1

source ./lib/functions.sh
source ./lib/binaries.sh
```

Comment (mflendrich) [reply setup.sh]: Thanks, it's a nice way to set a predictable working directory for a script. Updated to match the above.

**Comment by mflendrich** [comment]
:scream_cat: it will probably be easier to drop this PR and recreate after the other ones are merged.

@stealthybox is there an ETA of the aforementioned merges being completed?

**Comment by mflendrich** [comment]
@dlespiau 
Apparently all the conflicting PRs have been merged. I've recreated this change.
This time I split `binaries.sh` into separate per-binary files that are now very similar to each other.

PTanotherL.

**Pull request merged**